src/pep8ext_naming.py

```diff
@@ -437,6 +437,6 @@ def _extract_names(assignment_target):
 
 def is_mixed_case(name):
     stripped_name = name.strip('_')
-    head = stripped_name[:1]
-    tail = stripped_name[1:]
-    return head.islower() and tail and not (tail.islower() or tail.isdigit())
+    head_is_lower = stripped_name[:1].islower()
+    tail_has_upper = any(map(str.isupper, stripped_name[1:]))
```

Comment:
On second thought, I don't like it that this is not lazy-evaluated. I'll fix it with the next commit.